client/extensions/woocommerce/state/sites/products/actions.js

```diff
@@ -157,6 +159,16 @@ export function fetchProducts( siteId, params, successAction, failureAction ) {
 }
 
 export function productsUpdated( siteId, params, products, totalPages, totalProducts ) {
+	// This passed through the API layer successfully, but failed at the remote site.
```

Comment:
Could we add a unit test for this scenario?

Reply:
Added :) I put it at the data layer, so we know that the correct action is fired based on the API response.